File: app/src/main/java/com/six/cat/sixcat/fragment/HomeFragment.java
package com.six.cat.sixcat.fragment;

import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.support.design.widget.TabLayout;

import com.six.cat.sixcat.R;
import com.six.cat.sixcat.adapter.HomeFgAdapter;
import com.six.cat.sixcat.base.BaseRxLazyFragment;
import com.six.cat.sixcat.utils.SettingUtil;
import com.six.cat.sixcat.utils.ShowToast;
import com.trello.rxlifecycle2.LifecycleTransformer;

import java.util.List;

import butterknife.BindView;

/**
 * @author liguoying
 * @date 2017/12/11.
 */

public class HomeFragment extends BaseRxLazyFragment {
    private static HomeFragment instance;
    @BindView(R.id.vp_home_fragment)
    ViewPager mHomeViewPage;
    @BindView(R.id.ll_home_fragment_header_layout)
    LinearLayout mHomell;
    @BindView(R.id.tb_home_fragment)
    TabLayout mHomeTb;
    @BindView(R.id.iv_home_fragment_add_channel)
    ImageView mHomeIv;


    public static HomeFragment newInstance() {
        if (instance == null) {
            instance = new HomeFragment();
        }
        return instance;
    }

    @Override
    public int getLayoutResId() {
        return R.layout.fragment_home;
    }

    @Override
    public void finishCreateView(Bundle state) {
        initViewPage();
    }

    private void initViewPage() {
        HomeFgAdapter homeFgAdapter = new HomeFgAdapter(getChildFragmentManager(), getApplicationContext());
        mHomeViewPage.setOffscreenPageLimit(3);
        mHomeViewPage.setAdapter(homeFgAdapter);
        mHomeTb.setupWithViewPager(mHomeViewPage);
        mHomeTb.setTabMode(TabLayout.MODE_SCROLLABLE);
        mHomeIv.setOnClickListener(view -> ShowToast.shortTime("i click it"));
        mHomell.setBackgroundColor(SettingUtil.getInstance().getColor());
    }

    @Override
    public void onShowLoading() {

    }

    @Override
    public void onHideLoading() {

    }

    @Override
    public void onShowNetError() {

    }

    @Override
    public void setPresenter(Object presenter) {

    }

    @Override
    public void onShowNoMore() {

    }

    @Override
    public LifecycleTransformer bindToLife() {
        return null;
    }


}

File: app/src/main/java/com/sixcat/views/fragment/HomeFragment.kt
package com.sixcat.views.fragment

import com.google.android.material.tabs.TabLayout
import com.sixcat.R
import com.sixcat.adapter.HomeFgAdapter
import com.sixcat.base.BaseRxLazyFragment
import com.sixcat.utils.SettingUtil
import kotlinx.android.synthetic.main.fragment_home.*

/**
 * Author：Administrator
 * Data: 2018/9/4  11:45
 */

open class HomeFragment : BaseRxLazyFragment() {

    companion object {
        fun newInstance(): HomeFragment = HomeFragment()
    }

    private val homeFgAdapter by lazy { HomeFgAdapter(childFragmentManager, activity) }

    override fun getLayoutResId() = R.layout.fragment_home

    override fun initView() {
        initViewPage()
    }

    fun initViewPage() {
        vpHome.adapter = homeFgAdapter
        vpHome.offscreenPageLimit = 3
        tabHome.setupWithViewPager(vpHome)
        tabHome.tabMode = TabLayout.MODE_SCROLLABLE
        tabHome.setBackgroundColor(SettingUtil.instance.color)
    }

    override fun lazyLoad() {
    }

}